### PR TITLE
Update xtensa fully_connected kernel to call into xa_nnlib.

### PR DESCRIPTION
**Pull request by advaitjain**
This is the relevant code snippet from https://github.com/tensorflow/tensorflow/pull/44581 that is needed to use xa_nnlib with the fully_connected kernel for fusion_f1.

Profiling the keyword benchmark shows a reduction of ~20,000 ticks for invoke (if we use the updated xa_nnlib from https://github.com/tensorflow/tensorflow/pull/44581)

```
make -f tensorflow/lite/micro/tools/make/Makefile TARGET=xtensa OPTIMIZED_KERNEL_DIR=xtensa TARGET_ARCH=fusion_f1 XTENSA_CORE=F1_190305_swupgrade test_keyword_benchmark -j8
```

gives:
```
InitializeKeywordRunner() took 279792 ticks (279 ms)
KeywordRunNIerations(1) took 151304 ticks (151 ms)
KeywordRunNIerations(10) took 1512547 ticks (1512 ms)
```

Also confirmed that the kernel_fully_connected_test passes:
```
make -f tensorflow/lite/micro/tools/make/Makefile TARGET=xtensa OPTIMIZED_KERNEL_DIR=xtensa TARGET_ARCH=fusion_f1 XTENSA_CORE=F1_190305_swupgrade test_kernel_fully_connected_test -j8
```

Progress towards http://b/177457688

### CODE REVIEW

**Comment by advaitjain** [comment]
tagging @pnikam-cad @nyadla-sys

**Review by advaitjain — COMMENTED** [review]
comments based on discussion with @njeffrie

File: tensorflow/lite/micro/kernels/xtensa/fully_connected.cc [hunk 1]
```diff
@@ -196,6 +196,40 @@ TfLiteStatus EvalQuantizedInt8(TfLiteContext* context, TfLiteNode* node,
...
                  tflite::micro::GetTensorShape(output),
                  tflite::micro::GetTensorData<int8_t>(output));
+#elif defined(FUSION_F1)
+  FullyConnectedParams op_params = FullyConnectedParamsQuantized(data);
+  // Weights will always be symmetric quantized since we only support int
+  // quantization.
+  TFLITE_DCHECK(op_params.weights_offset == 0);
+
+  const RuntimeShape& output_shape = tflite::micro::GetTensorShape(output);
+  const int num_batches = output_shape.Dims(0);
```
[the remark this answers is not in the document]
Comment (advaitjain) [reply tensorflow/lite/micro/kernels/xtensa/fully_connected.cc:206]: done

File: tensorflow/lite/micro/kernels/xtensa/fully_connected.cc [hunk 2]
```diff
@@ -196,6 +196,40 @@ TfLiteStatus EvalQuantizedInt8(TfLiteContext* context, TfLiteNode* node,
                  tflite::micro::GetTensorData<int32_t>(bias),
                  tflite::micro::GetTensorShape(output),
                  tflite::micro::GetTensorData<int8_t>(output));
+#elif defined(FUSION_F1)
+  FullyConnectedParams op_params = FullyConnectedParamsQuantized(data);
+  // Weights will always be symmetric quantized since we only support int
+  // quantization.
+  TFLITE_DCHECK(op_params.weights_offset == 0);
```
[the remark this answers is not in the document]
Comment (advaitjain) [reply tensorflow/lite/micro/kernels/xtensa/fully_connected.cc:203]: done

**Comment by advaitjain** [comment]
@njeffrie, ready for review again.

**Review by njeffrie — APPROVED** [review]

**Pull request merged**